polynote-runtime/src/main/scala/polynote/runtime/macros/StructDataEncoderMacros.scala

```diff
@@ -105,17 +105,45 @@ class StructDataEncoderMacros(val c: whitebox.Context) {
                 cq"${name.decodedName.toString} => Some((((value: $A) => value.$name), ${encoderVariable.name}))"
             } :+ cq"_ => None"
 
+            val displayPrefix = A.typeSymbol.name.decodedName.toString
+            def displayField(name: TermName, stringifier: Tree, last: Boolean): Tree = q"""
+                  sb.append("  ")
```

Comment:
If `sb` was renamed below would that be caught at compile time or only at runtime? Would it be useful to pass in `"sb"` to `displayField` so it's clear where it comes from (and so a rename wouldn't break the code)?

Reply:
This is a quasiquote, so `sb` can't possibly be named different at runtime. It's part of a macro, which happens at compile-time (granted, a notebook blurs those lines... but it's safe to refer directly to `sb` here).

If the name is confusing, though, it can be changed.

Reply:
The name is fine. I'm not suggesting a change, I'm mainly asking to improve my knowledge of how this macro stuff works. The answer would probably be obvious if I knew more about quasiquotes. 

What I'm wondering is: say I renamed `sb` to `stringBuilder` on lines 131-136, but forgot to update this method. What would the failure mode be? Would it fail to compile? Or would it be caught when CI runs `CollectionReprsSpec` (which is what I meant by "runtime" in this case)?

Reply:
That's a good point. It's a little shredded across the functions here. I see what you were saying now about passing in the name to be used.

To answer your question, it would fail at compile-time of this macro. That happens at runtime of the spec (to your point), but would happen at test compile time if we were directly testing these macros somewhere. Which we aren't, and we probably should be.